services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/notifications/_notifications_ports.py

```diff
@@ -0,0 +1,78 @@
+from dataclasses import dataclass
+
+from fastapi import FastAPI
+from models_library.api_schemas_dynamic_sidecar.ports import InputStatus, OutputStatus
+from models_library.projects import ProjectID
+from models_library.projects_nodes_io import NodeID
+from models_library.services_types import ServicePortKey
+from models_library.users import UserID
+
+from ._notifier import Notifier
+
+
+@dataclass
+class PortNotifier:
+    app: FastAPI
+    user_id: UserID
+    project_id: ProjectID
+    node_id: NodeID
+
+    async def _send_output_port_status(
+        self, port_key: ServicePortKey, status: OutputStatus
+    ) -> None:
+        notifier: Notifier = Notifier.get_from_app_state(self.app)
```

Comment:
thought: Would this not be more generic if it would not depend on the app? you would need to pass the notifier.